be/src/exec/iceberg/iceberg_delete_builder.cpp

```diff
@@ -175,6 +180,66 @@ Status ORCEqualityDeleteBuilder::build(const std::string& timezone, const std::s
     return Status::OK();
 }
 
+Status ParquetEqualityDeleteBuilder::build(const std::string& timezone, const std::string& delete_file_path,
+                                           int64_t file_length,
+                                           const std::shared_ptr<DefaultMORProcessor> mor_processor,
+                                           std::vector<SlotDescriptor*> slot_descs,
+                                           TupleDescriptor* delete_column_tuple_desc,
+                                           const TIcebergSchema* iceberg_equal_delete_schema,
+                                           RuntimeState* state) {
+    std::unique_ptr<RandomAccessFile> file;
+    ASSIGN_OR_RETURN(file, _fs->new_random_access_file(delete_file_path));
+
+    std::unique_ptr<parquet::FileReader> reader;
+    try {
+        reader = std::make_unique<parquet::FileReader>(state->chunk_size(), file.get(), file->get_size().value(), 0);
+    } catch (std::exception& e) {
+        const auto s =
+                strings::Substitute("ParquetEqualityDeleteBuilder::build create parquet::FileReader failed. reason = $0", e.what());
+        LOG(WARNING) << s;
+        return Status::InternalError(s);
+    }
+
+    auto scanner_ctx = std::make_unique<HdfsScannerContext>();
+    auto scan_stats = std::make_unique<HdfsScanStats>();
+    std::vector<HdfsScannerContext::ColumnInfo> columns;
```

Comment:
does HdfsScannerContext::ColumnInfo implement move ctor? if not so, suggest to coding like this:
```cpp
use ColumnInfoPtr = std::shared_ptr<HdfsScannerContext::ColumnInfo>;
use ColumnInfos = std::vector<ColumnInfoPtr>;
```
use shared_ptr instead of bare type.